rswag-specs/lib/rswag/specs/example_group_helpers.rb

```diff
@@ -7,14 +7,18 @@ module Specs
     module ExampleGroupHelpers
       ActiveSupport::Deprecation.warn('Rswag::Specs: WARNING: Support for Ruby 2.6 will be dropped in v3.0') if RUBY_VERSION.start_with? '2.6'
 
-      def path(template, metadata = {}, &block)
+      def path(template, *metadata, &block)
+        metadata = _parse_metadata(metadata)
         metadata[:path_item] = { template: template }
         describe(template, metadata, &block)
       end
 
       [:get, :post, :patch, :put, :delete, :head, :options, :trace].each do |verb|
-        define_method(verb) do |summary, **metadata, &block|
-          api_metadata = { operation: { verb: verb, summary: summary } }.deep_merge(metadata)
+        define_method(verb) do |summary, *metadata, &block|
```

Comment:
I wonder... this kinda rebuilds RSpec's handling of parameters. Judging by [`describe`'s signature](https://github.com/rspec/rspec-core/blob/1eeadce5aa7137ead054783c31ff35cbfe9d07cc/lib/rspec/core/example_group.rb#L247-L275) in rspec-core, this could also work if we just pass on `*metadata` as second parameter to `describe`. Do the tests still pass if you do that?

Reply:
@franzliedke I don't know about tests failing, but that approach would fail if the extra metadata provided by the user was in hash form.

However, you don't need to parse it. Since none of rswag's metadata is in the set-if-true style, all you would need to do is something like this:
```ruby
define_method(verb) do |summary, *args, &block|
  metadata = args.last.is_a?(Hash) ? args.pop : {}
  api_metadata = { operation: { verb: verb, summary: summary } }.deep_merge(metadata)
  describe(verb, *args, **api_metadata, &block)
end
```
No parsing needed.